/*========================== begin_copyright_notice ============================

Copyright (C) 2025 Intel Corporation

SPDX-License-Identifier: MIT

============================= end_copyright_notice ===========================*/

diff --git a/GenXIntrinsics/lib/GenXIntrinsics/GenXSPIRVWriterAdaptor.cpp b/GenXIntrinsics/lib/GenXIntrinsics/GenXSPIRVWriterAdaptor.cpp
index 74efc92..b0d59bf 100755
--- a/GenXIntrinsics/lib/GenXIntrinsics/GenXSPIRVWriterAdaptor.cpp
+++ b/GenXIntrinsics/lib/GenXIntrinsics/GenXSPIRVWriterAdaptor.cpp
@@ -551,7 +551,7 @@ static void rewriteKernelArguments(Function &F) {
   for (auto &&ArgPair : llvm::zip(F.args(), NewF->args()))
     rewriteArgumentUses(InsPt, std::get<0>(ArgPair), std::get<1>(ArgPair));

-#if VC_INTR_LLVM_VERSION_MAJOR >= 17
+#if VC_INTR_LLVM_VERSION_MAJOR >= 17 && !defined(IGC_LLVM_TRUNK_REVISION)
   // There might be module level named metadata referencing old function, so replace those usages with new function.
   // This can be done safely (will not cause type mismatch) when only opaque pointers are used (since LLVM 17).
   F.replaceAllUsesWith(NewF);
